h5netcdf/tests/test_h5netcdf.py

```diff
@@ -574,6 +574,12 @@ def test_decode_string_error(tmp_local_or_remote_netcdf):
             assert ds.name == "/"
 
 
+def test_mode_warning(tmp_local_or_remote_netcdf):
+    with pytest.warns(FutureWarning):
+        with h5netcdf.File(tmp_local_or_remote_netcdf) as ds:
+            pass
```

Comment:
Seems `flake8` doesn't like that `ds` isn't used. We could `assert ds` instead of `pass`.

Reply:
What if we did
```python
with pytest.warns(FutureWarning):
    h5netcdf.File(tmp_local_or_remote_netcdf).close()
```

Reply:
Let's just duplicate the code of the two tests above. But I have no strong feelings what to choose. It should just work.